java-connectors/kafka-connect-azure-servicebus/src/test/java/io/lenses/streamreactor/connect/azure/servicebus/source/AzureServiceBusSourceConnectorTest.java

```diff
@@ -0,0 +1,136 @@
+  private static final String EMPTY_STRING = "";
+  private static final String JARMANIFEST_VERSION = "1.2.3";
+  private static final String JARMANIFEST_STRING = "some JarManifest String";
+  private AzureServiceBusSourceConnector testObj;
+  private JarManifest jarManifest;
+
+  @BeforeEach
+  void setUpAll() {
+    try (MockedConstruction<JarManifest> ignored = Mockito.mockConstruction(JarManifest.class)) {
+      testObj = new AzureServiceBusSourceConnector();
+      jarManifest = ignored.constructed().get(0);
+    }
+  }
+
+  @Test
+  void start() {
+    //given
+    when(jarManifest.buildManifestString()).thenReturn(JARMANIFEST_STRING);
+    Map<String, String> validProperties = buildValidProperties();
+
+    //when
+    testObj.start(validProperties);
+
+    //then
+    verify(jarManifest).buildManifestString();
+  }
+
+  @Test
+  void taskClassShouldReturnTaskClass() {
+    //when
+    Class<? extends Task> taskClass = testObj.taskClass();
+
+    //then
+    AzureServiceBusSourceTask.class.equals(taskClass);
+  }
+
+  @Test
+  void taskConfigsShouldReturnSameConfigForSimplePropertiesWithOneKcqlAndTwoMaxTasks() {
+    //given
+    Map<String, String> validProperties = buildValidProperties();
+    int maxTasksForThisScenario = 2;
+
+    //when
+    testObj.start(validProperties);
+    List<Map<String, String>> taskConfigs = testObj.taskConfigs(maxTasksForThisScenario);
+
+    //then
+    assertThat(taskConfigs).hasSize(2);
+    assertThat(taskConfigs.get(0)).isEqualTo(validProperties);
+    assertThat(taskConfigs.get(1)).isEqualTo(validProperties);
+  }
+
+  @Test
+  void configShouldReturnConfigClass() {
+    //when
+    ConfigDef configDef = testObj.config();
+
+    //then
+    AzureServiceBusSourceConfig.getConfigDefinition().equals(configDef);
+  }
+
+  @Test
+  void versionShouldReturnJarManifestsVersion() {
+    //given
+    when(jarManifest.getVersion()).thenReturn(JARMANIFEST_VERSION);
+
+    //when
+    String version = testObj.version();
+
+    //then
+    assertThat(version).isEqualTo(JARMANIFEST_VERSION);
+  }
+
+  private Map<String, String> buildValidProperties() {
+    String simpleKcql = buildKcqlWithNecessaryProperties();
+    HashMap<String, String> propertyMap = new HashMap<>();
+
+    propertyMap.put(AzureServiceBusConfigConstants.CONNECTOR_NAME, CONFIG_CONNECTOR_NAME);
+    propertyMap.put(AzureServiceBusConfigConstants.CONNECTION_STRING, EMPTY_STRING);
+    propertyMap.put(AzureServiceBusConfigConstants.KCQL_CONFIG, simpleKcql);
+
+    return propertyMap;
+  }
+
+  private String buildKcqlWithNecessaryProperties() {
+    final StringBuilder kcqlBuilder = new StringBuilder("INSERT INTO output SELECT * FROM input ");
+    kcqlBuilder.append("PROPERTIES(");
+
+    for (ServiceBusKcqlProperties property : ServiceBusKcqlProperties.values()) {
+      kcqlBuilder.append("'" + property.getPropertyName() + "'='" + property.getPropertyName() + "',");
```

Comment:
Let's not mix StringBuilders and String concatenation, picking one or the other and sticking to it would be better.

Reply:
Agreed!